src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -192,9 +192,15 @@ public function findByUrl($url, $webspaceKey, $locale = null)
 
     public function findByPage(UuidBehavior $page)
     {
+        $descendentPaths = [];
+        foreach ($this->webspaceManager->getWebspaceCollection() as $webspace) {
+            $descendentPaths[] = sprintf('ISDESCENDANTNODE(a, "/cmf/%s/custom-urls")', $webspace->getKey());
+        }
```

Comment:
Can we not just get the webspace from the page? I don't think that cross webspace custom urls are possible.

Reply:
could try that if @rgericke-elb can approve the performance improvement